packages/m365_defender/data_stream/event/elasticsearch/ingest_pipeline/pipeline_alert.yml

```
@@ -1,345 +1,448 @@
 ---
 description: Pipeline for processing Alert Event logs.
 processors:
+  # Event categorization
   - set:
       field: event.kind
-      value: [alert]
-  - set:
+      value: alert
+      tag: set_event_kind
+  - append:
       field: event.category
-      value: [file]
-      if: ctx.json?.properties?.EntityType?.toLowerCase() == 'file'
-  - set:
+      value: file
+      tag: append_event_category_file
+      if: ctx.json?.properties?.EntityType != null && ctx.json?.properties?.EntityType?.toLowerCase() == 'file'
```

```suggestion
      if: ctx.json?.properties?.EntityType != null && ctx.json.properties.EntityType.toLowerCase() == 'file'
```

We could probably skip the secondary checks here.